### PR TITLE
Move console redirect into wandb_run, add first pass of wandb_reinit and join fixes

### PR DESCRIPTION
**Pull request by raubitsj**

### CODE REVIEW

File: wandb/sdk/wandb_init.py [hunk 1]
```diff
@@ -335,148 +247,64 @@ def _log_setup(self, settings):
...
     def init(self):
         s = self.settings
         config = self.config
 
+        if s.reinit:
+            if len(self._wl._global_run_stack) > 0:
+                if len(self._wl._global_run_stack) > 1:
+                    print(
+                        "If you want to track multiple runs concurrently in wandb you should use multi-processing not threads"  # noqa: E501
+                    )
```
Comment (vanpelt) [inline wandb/sdk/wandb_init.py:261]: Do we want a `termwarn` here?  In regular usage of reinit, this case shouldn't be hit right?

File: wandb/sdk/wandb_run.py [hunk 2]
```diff
@@ -314,6 +379,106 @@ def _display_run(self):
...
+                logger.info("redirect2")
+            except (OSError, AttributeError) as e:
+                logger.error("failed to redirect", exc_info=e)
+            return
+
+        return
+
+        # redirect stdout
+        if platform.system() == "Windows":
+            lib_console.win32_redirect(stdout_slave_fd, stderr_slave_fd)
```
Comment (vanpelt) [inline wandb/sdk/wandb_run.py:410]: This is a noop currently right?  We may want a comment here or something to indicate what's happening in this case.

File: wandb/sdk/wandb_init.py (hunk 1 above)
Comment (raubitsj) [reply wandb/sdk/wandb_init.py:261]: fixed

File: wandb/sdk/wandb_run.py (hunk 2 above)
Comment (raubitsj) [reply wandb/sdk/wandb_run.py]: fixed

**Pull request merged**